### PR TITLE
docs: Update maintainer docs

### PR DESCRIPTION
**Pull request by simar7**
## Description

Adds info about upgrading dependencies prior to releasing.

## Checklist
- [x] I've read the [guidelines for contributing](https://trivy.dev/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://trivy.dev/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/community/maintainer/release-flow.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ For detailed behavior, please refer to [the GitHub Actions configuration][workfl
     Commits with prefixes like `chore` or `build` are not considered releasable, and no release PR is created.
     To include such commits in a release, you need to either include commits with `feat` or `fix` prefixes or perform a manual release as described [below](#manual-release-pr-creation).
 
+!!! tip
+    It's a good idea to check if there are any dependency updates created by dependabot waiting for your review.
+    If there are any, please review and merge them before creating a release. This will help to ensure that the release includes the latest dependencies and security patches.
```
[the remark this answers is not in the document]
Comment (simar7) [reply docs/community/maintainer/release-flow.md:17]: Hmm - while I agree with you that some amount of manual intervention is needed when upgrading dependencies as things can potentially break but I'm yet to find a scenario where our tests haven't caught such a thing.

Usually it's a test failure or a build failure when upgrading dependencies and so far I've noticed our tests have caught both such cases. There might be edge cases where integrations might be affected, such as another project that uses Trivy as a dependency. In such a case breakages could occur but it would be out of scope to test out in Trivy.

I updated this doc as I actually forgot to update some dependencies in this release (besides the security related ones). I realize it's not a big deal but I thought it's nice to have a release with all up to date dependencies. 

Let's hear from other members as well @aquasecurity/trivy what their thoughts are.

[the remark this answers is not in the document]
Comment (simar7) [reply docs/community/maintainer/release-flow.md:17]: > We encountered an issue with dependency updates in the past. IIRC, the problem was caused by a transitive dependency. #5807
> 

That's an interesting case that I did not consider. I think this could've been handled if we had a smoke test suite of tests that would test each dependency upgrade PR. I think it's worth considering adding such a set of tests given integration testing is as far as we get with testing in Trivy. We did something similar in trivy-checks to test each bundle with a release of Trivy: https://github.com/aquasecurity/trivy-checks/blob/8a62ef93635fa246c342c0837600baff75264f28/integration/check_examples_test.go#L31 


> I also recall several instances where issues introduced by dependency library updates were overlooked during testing. Although the frequency of such problems has decreased as our test suite has expanded, I still feel uneasy about updating all dependency libraries right before a release. How about simply encouraging checks to ensure that no updates are needed to resolve vulnerabilities or for the release?

Sure we can start with this then. I'll update my PR accordingly.

[the remark this answers is not in the document]
Comment (simar7) [reply docs/community/maintainer/release-flow.md]: Updated https://github.com/aquasecurity/trivy/pull/8674/commits/488e4ac46133d6815d94e963317a16ca63b03aa0

[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/community/maintainer/release-flow.md]: The specific issue comes from the fact that Trivy currently does not have access to a test AWS account, making automated testing impossible. I'm not sure whether this problem could have been detected through tests using stubs like LocalStack, but errors related to actual environment connections are often difficult to cover in tests. For example, I recall a case where Trivy worked with AWS ECR but encountered an error when connecting to Docker Hub. To improve test coverage, it would be beneficial to set up these environments and introduce end-to-end testing.
